snuba/consumers/consumer_builder.py

```diff
@@ -165,7 +166,7 @@ def build_base_consumer(self) -> BatchingConsumer[KafkaPayload]:
         return self.__build_consumer(
             ConsumerWorker(
                 self.dataset,
-                producer=self.producer,
+                producer=ConfluentKafkaProducer(self.__kafka_producer_configuration),
```

Comment:
Reason for not using KafkaProducer here ? Is it because we do not have a codec for the ConsumerWorker ?